### PR TITLE
Generic Fading Script for all Objects

### LINKED ISSUE #92: Fade out throwable objects and then destroy

### PR DESCRIPTION
**Pull request by jurcaua**
Work on: #92 

Quite janky logic, and does not look well for all situations, especially for things like the medicine bottle which has some parts of it that are transparent (due to issues I do not understand at all, to do with rendering transparent objects generally being a problem in computer graphics and by extension, in Unity).
* Mostly just has a sort of "pop-in" when working with fading IN a transparent object, fading out may be passable.

### CODE REVIEW

**Review by chadleytan — APPROVED** [review]
Went through it and looks like there is nothing wrong to me!

File: Assets/Scripts/ObjectFader.cs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+        float currentAlphaPercentage = 0f;
+        while (currentAlphaPercentage <= 1f)
+        {
+            SetAllMaterialsLerpedTransparency(currentAlphaPercentage);
+
+            currentAlphaPercentage += fadeSpeed * Time.deltaTime;
+            yield return null;
+        }
+
```
Comment (justineechen) [inline Assets/Scripts/ObjectFader.cs]: Blank space after while  ? PICK KNIT!

**Review by justineechen — COMMENTED** [review]
everything else looks good, sorry about the nit pick :( you don't even need to change it!

File: Assets/Scripts/ObjectFader.cs (hunk 1 above)
Comment (jurcaua) [reply Assets/Scripts/ObjectFader.cs]: THANK U
i want all the pick nicks

**Pull request merged**